File: exerciciosc++/PesoIdeal.cpp
#include <iostream>
#include <iomanip>

int main() {
    double altura, pesoIdeal;
    char sexo;

    std::cin >> altura;
    std::cin >> sexo;


    if (altura<0) {
        std::cout << "Entrada inválida!" << std::endl;
    }
    else if(sexo != 'm' &&  sexo != 'f'){
        std::cout << "Entrada inválida!" << std::endl;
    }

    else {
        if (sexo == 'm') {
            pesoIdeal = (72.7 * altura) - 58;
        }
        else {
            pesoIdeal = (62.1 * altura) - 44.7;
        }
        std::cout << std::fixed << std::setprecision(1) << pesoIdeal << " kg" << std::endl;
    }

    
    return 0;
}

File: exerciciosjava/src/main/java/aprendendojava/PesoIdeal.java
package main.java.aprendendojava;

import java.util.Scanner;

public class PesoIdeal {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);
        double pesoIdeal;
        String alturaStr = entrada.next();
        double altura = Double.parseDouble(alturaStr);
        entrada.nextLine();
        String sexoStr = entrada.nextLine();
        char sexo = sexoStr.charAt(0);
        if (altura <= 0){
            System.out.println("Entrada inválida!");
        }
        else if(sexo != 'm' &&  sexo != 'f'){
            System.out.println("Entrada inválida!");
        }
        else{
            if(sexo == 'm'){
                pesoIdeal = (72.7 * altura)-58;
            }
            else {
                pesoIdeal = (62.1 * altura)-44.7;
            }
            System.out.printf("%.1f kg\n", pesoIdeal);
        }
        entrada.close();
    }
}